paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/source/ContinuousFileStoreSource.java

```diff
@@ -73,6 +75,13 @@ public SplitEnumerator<FileStoreSourceSplit, PendingSplitsCheckpoint> restoreEnu
             splits = checkpoint.splits();
         }
         StreamTableScan scan = readBuilder.newStreamScan();
+        if (context.metricGroup() != null) {
+            try {
+                ((InnerStreamTableScan) scan)
+                        .withMetricsRegistry(new FlinkMetricRegistry(context.metricGroup()));
+            } catch (UnsupportedOperationException ignore) {
```

Comment:
No need to catch `UnsupportedOperationException`. If such exception is thrown, we should implement the method, not ignore it.